### PR TITLE
DRAFT: Adding sentieon-bwamem

### PR DESCRIPTION
**Pull request by asp8200**
DRAFT

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/sarek.nf [hunk 1]
```diff
@@ -335,10 +335,13 @@ workflow SAREK {
     bwa                    = params.bwa                     ? Channel.fromPath(params.bwa).collect()       : PREPARE_GENOME.out.bwa
     bwamem2                = params.bwamem2                 ? Channel.fromPath(params.bwamem2).collect()   : PREPARE_GENOME.out.bwamem2
     dragmap                = params.dragmap                 ? Channel.fromPath(params.dragmap).collect()   : PREPARE_GENOME.out.hashtable
+    sentieon_bwamem        = params.sentieon_bwamem         ? Channel.fromPath(params.sentieon_bwamem).collect() : PREPARE_GENOME.out.sentieon_bwamem
```
Comment (maxulysse) [inline workflows/sarek.nf:338]: I think we can use the same `params.bwa`.
but maybe let's add a `params.sentieon` as an extra failsafe, similarly to what we had in sarek pre-3.0.

Comment (FriederikeHanssen) [reply workflows/sarek.nf:338]: are the indices the same?

File: conf/test/test.config [hunk 2]
```diff
@@ -58,6 +58,15 @@ params {
...
 
 process {
+    withLabel: 'sentieon' {
+        ext.sentieon_auth_mech_base64 = secrets.SENTIEON_AUTH_MECH_BASE64
+        ext.sentieon_auth_data_base64 = secrets.SENTIEON_AUTH_DATA_BASE64
+    }
+
+    withName: 'SENTIEON_BWAMEM' {
+        ext.args = "-R \"@RG\\tID:sample_lane\\tPU:lane\\tSM:patient_sample\\tLB:sample\\tDS:fasta\\tPL:seqplatform\""
+    }
```
Comment (maxulysse) [inline conf/test/test.config:68]: Can e get that properly set up in `conf/modules/aligner.config`?

File: workflows/sarek.nf (hunk 1 above)
Comment (asp8200) [reply workflows/sarek.nf:338]: > are the indices the same?

I got the impression from Don that Sentieon is just a wrapper for the regular bwa index command, but I'm not entirely sure. The md5sums of the produced index files are unchanged, which backs my impression.

File: conf/test/test.config (hunk 2 above)
Comment (asp8200) [reply conf/test/test.config:68]: > Can e get that properly set up in `conf/modules/aligner.config`?

Good idea. I'll give it a shot.

**Comment by maxulysse** [comment]
Closing as it's a duplicate of #994

**Pull request closed without merging**